Orange/widgets/data/owtranspose.py

```diff
@@ -0,0 +1,108 @@
+from Orange.data import Table, StringVariable
+from Orange.widgets.settings import (Setting, ContextSetting,
+                                     DomainContextHandler)
+from Orange.widgets.utils.itemmodels import DomainModel
+from Orange.widgets.widget import OWWidget, Msg
+from Orange.widgets import gui
+
+
+class OWTranspose(OWWidget):
+    name = "Transpose"
+    description = "Transpose data table."
+    icon = "icons/Transpose.svg"
+    priority = 2000
+
+    inputs = [("Data", Table, "set_data")]
+    outputs = [("Data", Table)]
+
+    resizing_enabled = False
+    want_main_area = False
+
+    settingsHandler = DomainContextHandler(metas_in_res=True)
+    feature_type = ContextSetting(0)
+    feature_names_column = ContextSetting(None)
+    auto_apply = Setting(True)
+
+    class Error(OWWidget.Error):
+        value_error = Msg("{}")
+
+    def __init__(self):
+        super().__init__()
+        self.data = None
+
+        # GUI
+        box = gui.vBox(self.controlArea, "Feature names")
+        self.feature_radio = gui.radioButtonsInBox(
+            box, self, "feature_type", callback=self.apply,
```

Comment:
One more thing (sorry!): changing the radio button always applies even if the auto-apply is off. Change `callback=self.apply` to `callback=lambda: self.apply()`.

(The problem is that `callback` is bound to the original `self.apply` before `gui.auto_commit` replaces it with the conditional apply.)